### PR TITLE
Reporting Dashboard

### PR DESCRIPTION
**Pull request by batemapf**
## Description

Address the primary reporting components of #519 by automating the weekly "18F Operations Dashboard" that @joshuabailes assembles each week. The core of this new functionality is a report page that details performance versus hourly and revenue targets for two time periods: (1) fiscal year to date (i.e. "cumulative"); and (2) for the requested reporting period (i.e. "point in time").

This change:

- Creates a new model within the hours app, Targets, that stores revenue and hour target info;

- Creates a new view within the hours app that displays summary information based on actual hours logged (via TimecardObject's) and Targets info and renders that view in a new dashboard template in straightforward (some might say overly simple) table form;

- Creates another new view within the hours app that lists all reports after a static date (this should be refactored later to be dynamic, likely based on on which reporting periods have a viable Targets') and renders this view via a new dashboard list template;

### Latest screen shots

Dashboard reports selection:
screen shot 2016-11-30 at 4 32 23 pm

Report detail:

s7bv46lme3

### CODE REVIEW

**Comment by maya** [comment]
@batemapf see my refactor of tables here: https://github.com/18F/tock/pull/588.

Needs the following:
- [ ] Add  `<thead>` and `<tbody>` elements to properly markup the table. 
- [ ] Add `scope="col"` or `scope="row"` to `th`s. See: https://standards.usa.gov/tables/

Consider adding units of measurement to the table.

**Comment by batemapf** [comment]
@maya OK if i turn this one over to you for front end work? i believe the backend dev work to be in a solid MVP state (with @annalee's 👀 of course).

**Comment by batemapf** [comment]
@maya per our convo and your comment above, i updated the template to meet USWDS. now i am _really_ turning it over to you!

**Comment by maya** [comment]
@batemapf Tables are looking good. Two small fixes noted here: https://github.com/18F/tock/pull/587/commits/165c0994b11cff741053ff654cee6baacf366360

**Comment by maya** [comment]
@batemapf it would be good if the word "hours" is noted somewhere in the hour data: either in the table title, table header, or data itself. Dollars are good bc we use the `$` sign.

**Comment by maya** [comment]
Ah, I just saw the `Hours` title above the table. I missed it the first time!

File: tock/tock/templates/hours/dashboard.html [hunk 1]
```diff
@@ -27,93 +27,108 @@
...
-    <td>{{ hours_billed_fytd }}</td>
-    <td>{{ variance_plan_fytd }} ({{ p_variance_plan_fytd }})</td>
-  </tr>
+  <thead>
+    <tr>
+      <th scope="col"></th>
+      <th scope="col">FYTD Target</th>
+      <th scope="col">FYTD Performance</th>
+      <th scope="col">Variance</th>
+    </tr>
```
Comment (batemapf) [inline tock/tock/templates/hours/dashboard.html]: Done.

File: tock/tock/templates/hours/dashboard.html [hunk 2]
```diff
@@ -27,93 +27,108 @@
...
+  <thead>
+    <tr>
+      <th></th>
+      <th scope="col">FYTD Target</th>
+      <th scope="col">FYTD Performance</th>
+      <th scope="col">Variance</th>
+    </tr>
+  </thead>
+  <tbody>
   <tr>
```
Comment (batemapf) [inline tock/tock/templates/hours/dashboard.html]: Done.

File: tock/tock/templates/hours/dashboard.html (hunk 1 above)
Comment (batemapf) [inline tock/tock/templates/hours/dashboard.html]: See https://github.com/18F/tock/pull/587/commits/58bb39cf634a1086cc59348f7d5db7e114691c91.

File: tock/tock/templates/hours/dashboard.html (hunk 2 above)
Comment (batemapf) [inline tock/tock/templates/hours/dashboard.html]: See https://github.com/18F/tock/pull/587/commits/58bb39cf634a1086cc59348f7d5db7e114691c91.

Comment (maya) [inline tock/tock/templates/hours/dashboard.html]: The whole code block below needs to be indented, not just that line. Sorry if that wasn't clear.

**Comment by batemapf** [comment]
OK, @annalee i believe you are up for engineer review!

**Pull request merged**